test/unit/node/proxy_agent.test.ts

```diff
@@ -0,0 +1,59 @@
+import { shouldEnableProxy } from "../../../src/node/proxy_agent"
+
+/**
+ * Helper function to set an env variable.
+ *
+ * Returns a function to cleanup the env variable.
+ */
+function setEnv(name: string, value: string) {
```

Comment:
Nit: given this is a specific test helper, it would be nice if we could move the `cleanup`  behavior into the `beforeEach` so that it may only have to be declared once. This is a bit like a React hook, and allows us to target a specific key, rather than dereferencing the entire `process.env` object. 

```ts
// test_helpers.ts 
function createSetEnv(key: string) {
  const initialValue = process.env[key]
  const setValue = (nextValue: string | undefined) => process.env[key] = nextValue
  const resetValue = () => process.env[key] => initialValue

  return [setValue, resetValue]
}

// test.ts

const [setHTTPProxy, resetHTTPproxy] = createSetEnv(“HTTP_PROXY”)
const [setNoProxy, resetNoProxy] = createSetEnv(“NO_PROXY”)
{
  beforeEach() {
    jest.resetModule()
    resetHTTPproxy()
    resetNoProxy()
  }
```

Reply:
Oh waaaay better than my solution. The original `cleanup()` I wrote was partially inspired by previous experience cleaning up `useEffect` functions. I really like your implementation. Thanks for writing that out! I'll make these changes.